python/imageutils.py

```diff
@@ -383,40 +384,109 @@ def reconstruct_slice_slice(self, reconstructed, legend, mem=None):
         with open(legend, "r") as legend:
 
             remaining_mem = mem
-            data = None                
+            data = None
 
             for slice_name in legend:
-                
+
                 if slice_name == "" or slice_name is None:
-                    continue                
+                    continue
 
                 slice_name = slice_name.strip()
                 print "Loading slice: {0}".format(slice_name)
-                
+
                 slice_img = self.load_image(slice_name)
                 slice_shape = slice_img.header.get_data_shape()
-                slice_bytes = slice_img.header.single_vox_offset + slice_img.header['bitpix']/8 * (slice_shape[0] * slice_shape[1] * slice_shape[2])
-
+                slice_bytes = slice_img.header.single_vox_offset + slice_img.header['bitpix'] / 8 * (
+                slice_shape[0] * slice_shape[1] * slice_shape[2])
 
                 if data is None:
                     data = slice_img.get_data()[...]
                 else:
-                    data = np.concatenate((data, slice_img.get_data()[...]), axis = 2)    
-                
+                    data = np.concatenate((data, slice_img.get_data()[...]), axis=2)
+
                 if remaining_mem is not None:
                     remaining_mem = remaining_mem - slice_bytes
 
                 if remaining_mem is None or remaining_mem / slice_bytes < 1:
                     write_start = time()
                     reconstructed.write(data.tobytes('F'))
-                    print "Write time for slices: {0}".format(time()-write_start)
-        
+                    print "Write time for slices: {0}".format(time() - write_start)
+
                     data = None
                     remaining_mem = mem
 
+    def reconstruct_mreads(self, reconstructed, legend, mem):
+        rec_dims = self.header.get_data_shape()
+        (Y_size, Z_size, X_size) = rec_dims
+        bytes_per_voxel = self.header['bitpix'] / 8
+        print "The shape of the orignal image is {}".format(rec_dims)
+
+        # extract all the info we needed to reconstruct from the legend file
+        with open(legend, 'r')  as legend:
+            filename_prefix = legend.readline().strip().split('/')[-1].split('_')[0]
+            filename_ext = legend.readline().strip().split(".", 1)[1]
+            blocks_dir = legend.readline().strip().rsplit('/', 1)[0]
+            block_shape = nib.load(legend.readline().strip()).header.get_data_shape()
+
+            print "filename_prefix: {0}, filename_ext: {1}".format(filename_prefix, filename_ext)
+            print "blocks located in {0}, block's shape is {1}".format(blocks_dir, block_shape)
+
+            legend.seek(0, 0)
+            for i, l in enumerate(legend):
+                pass;
```

Comment:
these lines is just to calculate how many blocks per axis. If not work for Bigbrain, I will figure out another way